app/src/main/java/com/android/sample/ui/activity/CreateActivity.kt

```diff
@@ -202,381 +181,99 @@ fun CreateActivityScreen(
-                  itemsList = selectedImages,
-                  deleteImage = { bitmap -> selectedImages.remove(bitmap) })
-              Spacer(modifier = Modifier.height(STANDARD_PADDING.dp))
-
-              RemainingPlace(title, maxTitleLength)
-              OutlinedTextField(
-                  value = title,
-                  onValueChange = {
-                    if (it.length <= maxTitleLength) {
-                      title = it
-                    }
+              ActivityForm(
+                  context = context,
+                  selectedImages = selectedImages,
+                  onOpenDialogImage = { showDialogImage = true },
+                  onDeleteImage = { bitmap -> selectedImages.remove(bitmap) },
+                  title = title,
+                  onTitleChange = { title = it },
+                  maxTitleSize = maxTitleLength,
+                  description = description,
+                  maxDescriptionSize = maxDescriptionLength,
+                  onDescriptionChange = { description = it },
+                  onClickDate = { dateIsOpen = true },
+                  onCloseDate = { dateIsOpen = false },
+                  onSelectDate = {
+                    dueDate = it
+                    dateIsOpen = false
+                    dateIsSet = true
                   },
-                  label = { Text("Title") },
-                  modifier =
-                      Modifier.padding(STANDARD_PADDING.dp)
-                          .fillMaxWidth()
-                          .testTag("inputTitleCreate"),
-                  placeholder = {
-                    Text(text = stringResource(id = R.string.request_activity_title))
+                  dueDate = dueDate,
+                  dateIsOpen = dateIsOpen,
+                  dateIsSet = dateIsSet,
+                  onClickStartingTime = { startTimeIsOpen = true },
+                  startTimeIsOpen = startTimeIsOpen,
+                  startTimeIsSet = startTimeIsSet,
+                  onStartTimeSelected = { time ->
+                    startTime =
+                        time.toInstant().atZone(ZoneId.systemDefault()).toLocalTime().toString()
+                    startTimeIsOpen = false
+                    startTimeIsSet = true
                   },
-                  singleLine = true,
-              )
-
-              Spacer(modifier = Modifier.height(STANDARD_PADDING.dp))
-              RemainingPlace(description, maxDescriptionLength)
-              OutlinedTextField(
-                  value = description,
-                  onValueChange = {
-                    if (it.length <= maxDescriptionLength) {
-                      description = it
-                    }
+                  startTime = startTime,
+                  onCloseStartTime = { startTimeIsOpen = false },
+                  onClickDurationTime = { durationIsOpen = true },
+                  durationIsOpen = durationIsOpen,
+                  durationIsSet = durationIsSet,
```

Comment:
It would be great if you can add comments to the variables that are not very easy to understand like  durationIsOpen and  durationIsSet

Reply:
Working on it!